pkg/asset/installconfig/azure/session.go

```diff
@@ -224,11 +235,17 @@ func newSessionFromCredentials(cloudEnv azureenv.Environment, credentials *Crede
 		AADEndpoint:  cloudEnv.ActiveDirectoryEndpoint,
 	}
 	c.Resource = cloudEnv.TokenAudience
-	authorizer, err := c.Authorizer()
+
+	cred, err := azidentity.NewClientSecretCredential(credentials.TenantID, credentials.ClientID, credentials.ClientSecret, nil)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to get client credentials authorizer")
+		return nil, errors.Wrap(err, "failed to get client credentials secret")
 	}
 
+	scopes := []string{endpointToScope(cloudEnv.ResourceManagerEndpoint)}
+	// Use an adapter so azidentity in the Azure Management libraries can be
+	// used in place of Authorizer when calling Azure Management APIs.
+	authorizer := azidext.NewTokenCredentialAdapter(cred, scopes)
```

Comment:
The error message you're seeing now in CI is because no tenant is being provided. Instead of using `azidext.NewTokenCredentialAdapter` try using `azidext.NewDefaultAzureCredentialAdapter`. The `*azidentity.DefaultAzureCredentialOptions` will take the tenantID and I _think_ `AuthorityHost` should be set to `cloudEnv.ActiveDirectoryEndpoint`.

Reply:
The problem with the `azidext.NewDefaultAzureCredentialAdapter` is that it uses the `azidentity.NewDefaultAzureCredential` which, according to the docs,
 ```
It attempts to authenticate with each of these credential types, in the following order, stopping when one provides a token: - EnvironmentCredential - ManagedIdentityCredential - AzureCLICredential
```
So I don't see a way for us to pass in the client credentials through code.

Reply:
You are correct, my mistake! You are indeed already passing the tenant to `NewClientSecretCredentials`, but   `AuthorityHost` is nil. So I think we need to figure out what `AuthorityHost` should be, perhaps `cloudEnv.ActiveDirectoryEndpoint`.